xlive/XLive/UserManagement/CUser.cpp

```diff
@@ -204,13 +209,12 @@ void CUserManagement::ConfigureLocalUser(XNADDR* pxna, ULONGLONG xuid, char* use
 	memset(&local_user, NULL, sizeof(CUser));
 	memcpy(&local_user.xnaddr, pxna, sizeof(XNADDR));
 	local_user.secure.s_addr = pxna->inaOnline.s_addr;
-	this->secure_packet = new char[12 + sizeof(XNADDR)];
-	memset(secure_packet, NULL, 12 + sizeof(XNADDR));
+	memset(&securePacket, NULL, sizeof(SecurePacket));
```

Comment:
using `0` instead of `NULL` or the `ZeroMemory` macro would make it more clear. `NULL` could make it seem like you are dealing with an array of pointers.